plugin/client_rpc.go

```diff
@@ -363,6 +367,108 @@ func (s *hooksRPCServer) ServeHTTP(args *Z_ServeHTTPArgs, returns *struct{}) err
 	return nil
 }
 
+type Z_PluginHTTPArgs struct {
```

Comment:
2/5 - I'd just buffer the request and response and marshal them across as []byte. The majority of use cases for this will be small, streaming requires an extra TCP connection back and a then separate roundtrips for the packets. Sending []bytes would be more performant, and simplify the code considerably.